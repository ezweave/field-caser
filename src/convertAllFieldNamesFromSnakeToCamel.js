import { camelCase } from 'lodash';
import { deepFieldTransformer } from './deepFieldTransformer';
export const convertAllFieldNamesFromSnakeToCamel = deepFieldTransformer(camelCase);