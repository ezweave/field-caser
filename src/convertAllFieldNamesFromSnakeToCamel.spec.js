import { convertAllFieldNamesFromSnakeToCamel } from './convertAllFieldNamesFromSnakeToCamel';

const jsonWithSnakeCaseFields = require('./__fixtures/jsonWithSnakeCaseFields.json');

describe('convertAllFieldNamesFromSnakeToCamel', () => {
  it('converts nested field names to camelcase', () => {
    expect(convertAllFieldNamesFromSnakeToCamel(jsonWithSnakeCaseFields)).toMatchSnapshot();
  });
});