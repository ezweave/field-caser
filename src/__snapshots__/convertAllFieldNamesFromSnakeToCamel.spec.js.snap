// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertAllFieldNamesFromSnakeToCamel converts nested field names to camelcase 1`] = `
Object {
  "details": Object {
    "cars": Object {
      "0": Object {
        "brand": "Pontiac",
        "model": "Aztek",
        "yearOfManufacture": "2004",
      },
      "1": Object {
        "brand": "Saturn",
        "model": "Vue",
        "yearOfManufacture": "2004",
      },
    },
    "dateOfBirth": "1/1/1970",
  },
  "firstName": "Joe",
  "lastName": "Consumer",
}
`;
